src/main/java/com/conveyal/gtfs/loader/JdbcTableWriter.java

```diff
@@ -0,0 +1,542 @@
+            }
+            for (Table referencingTable : referencingTables) {
+                Table parentTable = referencingTable.getParentTable();
+                if (parentTable != null && parentTable.name.equals(specTable.name) || referencingTable.name.equals("shapes")) {
+                    // If a referencing table has the current table as its parent, update child elements.
+                    JsonElement childEntities = jsonObject.get(referencingTable.name);
+                    if (childEntities == null) {
+                        throw new SQLException(String.format("Child entities %s must not be null", referencingTable.name));
+                    }
+                    int entityId = isCreating ? (int) newId : id;
+                    updateChildTable(childEntities.getAsJsonArray(), entityId, isCreating, referencingTable, connection);
+                }
+            }
+            // Add new ID to JSON object.
+            jsonObject.addProperty("id", newId);
+            // FIXME: Should this return the entity from the database?
+            return jsonObject.toString();
+        } catch (SQLException e) {
+            LOG.error("Error updating entity", e);
+            DbUtils.closeQuietly(connection);
+            throw e;
+        }
+    }
+
+    /**
+     * Creates a prepared statement for an entity update or create operation. Uses a map of Field objects with values to
+     * construct SQL statement for the given table name.
+     */
+    private static PreparedStatement createPreparedUpdate(Integer id, boolean isCreating, Map<Field, String> fieldStringMap, String tableName, Connection connection, boolean batch) throws SQLException {
+        String statementString;
+        // Collect field names for string joining from JsonObject.
+        List<String> fieldNames = fieldStringMap.keySet().stream()
+                // If updating, add suffix for use in set clause
+                .map(field -> isCreating ? field.getName() : field.getName() + " = ?")
+                // Ensure fields are set according to alphabetical order of field name
+                // This order MUST coincide with whatever function is used to fill values (see fillValues)
+                .sorted()
+                .collect(Collectors.toList());
+        if (isCreating) {
+            // If creating a new entity, use default next value for ID.
+            String questionMarks = String.join(", ", Collections.nCopies(fieldNames.size(), "?"));
+            statementString = String.format("insert into %s (id, %s) values (DEFAULT, %s)", tableName, String.join(", ", fieldNames), questionMarks);
+        } else {
+            // If updating an existing entity, specify record to update fields found with where clause on id.
+            statementString = String.format("update %s set %s where id = %d", tableName, String.join(", ", fieldNames), id);
+        }
+        // Prepare statement and set RETURN_GENERATED_KEYS to get the entity id back
```

Comment:
This comment mostly repeats words present in the next statement. It might be better to explain the "why", for example: "Set the RETURN_GENERATED_KEYS flag on the PreparedStatement because it may be creating new rows, in which case we need to know the auto-generated IDs of those new rows."